Animals

sjtsilvester opened the pull request.

Commit:
[WIP] testing animals

Commit:
Basic animal and component system

Commit:
Add animal spawning system

Commit:
Animals detect animals in their range, identify food, and hunt food in line of sight

Commit:
Animals can eat each other

Commit:
Breeding behaviour in entities without the breeding

Commit:
Add breeding and clean up rest of repo.

Commit:
Add TODO comment highlighting the need to refact the behaviour class.

Commit:
Add more animals, improve AI, balance a little, add demo setup

Commit:
Added some things for demo

Commit:
[WIP] Refactor of animal behaviour code

Commit:
[WIP] More refactoring of animal behaviour

Commit:
Refactored entity system and animal behaviour logic

Commit:
Organise file structure

Commit:
Tidy up animal code and add interaction with atmosphere.

Commit:
Add values for animal/atmosphere interaction.

Commit:
Remove blend files

Commit:
Change scene files.

Commit:
Fix code style issues

Commit:
Fix uncorrect name of enum

Commit:
Animals procedurally generate with world

Commit:
Add scene file

Commit:
Case sensitive path names

Commit:
Add main scene and fix enum in spawner

Commit:
Animals can eat plants

ianyfan committed:
Fixed csproj from rebase

Commit:
Merge branch 'master' into animals

Commit:
Animals eat plants, can be eaten, and performance increase.

Plants don't disappear once eaten still.

Commit:
Merge branch 'master' into animals

Commit:
Removed hidden files

JamesHinshelwood reviewed (changes requested):
NullReferenceException when you click in mid air.
Maybe larger animals should drop more meat?
Also, didn't seem to see animals spawning as I wondered far away from home base - how is this meant to work and how can I test it?

Commit:
Bug fixes and changes to animal generation

Bigger animals drop more food, clicking on nothing doesn't break anything, procedurally generating animals dont' spawn in existing chunks
